weave/trace/op_extensions/log_once.py

```diff
@@ -0,0 +1,28 @@
+from typing import Callable
+
+LOG_ONCE_MESSAGE_SUFFIX = " (subsequent messages of this type will be suppressed)"
+logged_messages = []
+
+
+def log_once(log_method: Callable[[str], None], message: str) -> None:
```

Comment:
Is the intent to use this with many different loggers?

If just python logging, in future you can use the builtin `addFilter`